backend/inspirehep/assign/utils.py

```diff
@@ -71,5 +114,46 @@ def can_claim(data, author_profile_recid):
             for author_name in current_author_profile.get("name.name_variants", [])
         ]
     )
-    lit_author = get_author_by_recid(lit_record[0], int(author_profile_recid))
+    lit_author = get_author_by_recid(lit_record, int(author_profile_recid))
     return author_names & set([lit_author.get("full_name").split(",")[0]])
+
+
+def check_author_compability_with_lit_authors(literature_control_number):
```

Comment:
can we split the logic?

Reply:
we can, although imo we have 3 simple checks for 3 cases there